waifu2x-ncnn-vulkan: add page (#9478)
tldr-pages/tldr: pages/common/waifu2x-ncnn-vulkan.md

# waifu2x-ncnn-vulkan

> Image upscaler for manga/anime-style images using NCNN neural network framework.
> More information: <https://github.com/nihui/waifu2x-ncnn-vulkan>.

> - Upscale an image:

Reviewer: ```suggestion
- Upscale a specific image:
```

Optional fix: `a specific` may sound better.

`waifu2x-ncnn-vulkan -i {{path/to/input_file}} -o {{path/to/output_file}}`

> - Upscale an image by a custom scale factor and denoise it:

Reviewer: ```suggestion
- Upscale an image by a specific scale factor and denoise it:
```

The same issue as above.

`waifu2x-ncnn-vulkan -i {{path/to/input_file}} -o {{path/to/output_file}} -s {{1|2|4|8|16|32}} -n {{-1|0|1|2|3}}`

> - Save the upscaled image in a specific format:

Reviewer: ```suggestion
- Save an upscaled image in a specific format:
```

Optional fix: `an` may sound better.

# waifu2x-ncnn-vulkan

> Image upscaler for manga/anime-style images using NCNN neural network framework.

> > Runs fast on Intel/AMD/Nvidia/Apple-Silicon with Vulkan API.

Reviewer 2: ```suggestion
```
I think this is just unnecessary detail

> > Runs fast on Intel / AMD / Nvidia / Apple-Silicon with Vulkan API.

Reviewer 3: Is this line necessary?

Author: My intention was to emphasize the need of having Vulkan compatible GPU in order to run this. For that I coped the second line from project's description.

> > Runs fast on Intel / AMD / Nvidia / Apple-Silicon with Vulkan API.

Reviewer 3: ```suggestion
> Runs fast on Intel/AMD/Nvidia/Apple-Silicon with Vulkan API.
```
I think the space isn't necessary.

> More information: <https://github.com/nihui/waifu2x-ncnn-vulkan>.

> - Upscale image:

Reviewer 2: ```suggestion
- Upscale an image:
```

`waifu2x-ncnn-vulkan -i path/to/file -o path/to/output/file`

> - Upscale image by the scale factor of 4 and denoise it:

Reviewer 3: ```suggestion
- Upscale image by a custom scale factor and denoise it:
```

> - Upscale image by a custom scale factor and denoise it:

Reviewer 2: ```suggestion
- Upscale an image by a custom scale factor and denoise it:
```

Reviewer 2: what do we mean by denoise + it isn't a word

Author: From `--help`:
> -n noise-level       denoise level (-1/0/1/2/3, default=0)

This basically controls level of smoothing that happens with the image during upscaling. Higher value is more smoothing = less "noise" which can be e.g. artifacts from lossy compression.

> `waifu2x-ncnn-vulkan -i path/to/file -o path/to/output/file -s 4 -n 3`

Reviewer 3: ```suggestion
`waifu2x-ncnn-vulkan -i {{path/to/input_file}} -o {{path/to/output_file}} -s {{1/2/4/8/16/32}} -n {{-1/0/1/2/3}}`
```
I think it is better to list all the choices for scale factor and noise as it has limited options only.

- Save the upscaled image in a specific format:

> `waifu2x-ncnn-vulkan -i {{path/to/input_file}} -o {{path/to/output_file}} -f {{jpg/png/webp}}`

Reviewer 2: ```suggestion
`waifu2x-ncnn-vulkan -i {{path/to/input_file}} -o {{path/to/output_file}} -f {{jpg|png|webp}}`
```

small syntax error here, we use pipes to show options/choices
